src/pybind/mgr/restful/module.py

```diff
@@ -599,6 +602,14 @@ def submit_request(self, _request, **kwargs):
         with self.requests_lock:
             request = CommandsRequest(_request)
             self.requests.append(request)
+            if len(self.requests) > self.max_requests:
+                excess_requests = len(self.requests) - self.max_requests
+                self.log.info("Trimming %d requests due to exceeded maximum requests limit", len(self.requests) - self.max_requests)
+                trimmed_requests = self.requests[:excess_requests]
+                unfinished_trimmed = sum(1 for req in trimmed_requests if not req.is_finished())
+                if unfinished_trimmed:
+                    self.log.error("Trimmed %d unfinished requests due to exceeded maximum requests limit", unfinished_trimmed)
+                self.requests = self.requests[-self.max_requests:]
```

Comment:
two issues:

1. `self.max_requets` is initialized when the module is created. so we can just remove the first request which is finished, if none of them is finished, we pop one, instead of trimming the exceeding ones. like

    ```py
    if len(self.requests) > self.max_requests:
        req_to_trim = 0
        for i, req enumerate(self.requests):
            if req.is_finished():
                self.log.error("Trimmed one finished request due to exceeded maximum requests limit")
                req_to_trim = i
                break
        else:
            self.log.error("Trimmed the oldest unfinished request due to exceeded maximum requests limit")
        self.requests.pop(req_to_trim)
    ```
2. `self.requests = self.requests[-self.max_requests:]` is wrong. what you meant to put is probably `self.requests = self.requests[:-unfinished_trimmed]`.

anyway, i'd propose the first solution.

Reply:
@tchaikov right, fixing that